src/Lean/Elab/BuiltinCommand.lean

```diff
@@ -118,6 +118,21 @@ private partial def elabChoiceAux (cmds : Array Syntax) (i : Nat) : CommandElabM
 @[builtin_command_elab choice] def elabChoice : CommandElab := fun stx =>
   elabChoiceAux stx.getArgs 0
 
+/-- Declares a list of type universes.
```

Comment:
One answer to that question is "you need it when `autoImplicit` is `false`, which it is in mathlib".

When it's not, the answer is "it let's you choose the *order* of the universe variables.

Reply:
I don't know about the precis rules either. My experience is that I want type universe variables when I want to control precisely what the type are, most likely because there are relatively complex type universe constraints. Another one is when I want to be sure what universe a type is in.

Reply:
I think the broader question is whether this is the right place to have extensive discussions about these concepts. When we document `namespace` for instance, do we want a full explanation of the namespace system? I don't have an answer, one could argue it would be better to point to the relevant part of the reference document (type universes, namespaces...) and only provide a relatively short of the syntax here.

Reply:
For now, we don't have a comprehensive reference to point at, but once we do, I think the details clearly belong there. In the meantime, I don't think there's a need to be comprehensive here.